### PR TITLE
Update discover-granules

### PR DESCRIPTION
**Pull request by yjpa7145**
Extremely extensive changes in order to get the discover-granules task working with the message adapter, as well as fix bugs found along the way.

### CODE REVIEW

File: cumulus/tasks/discover-pdrs/package.json [hunk 1]
```diff
@@ -53,6 +53,8 @@
     "@ava/babel-preset-stage-4": "^1.1.0",
     "@ava/babel-preset-transform-test-files": "^3.0.0",
     "ava": "^0.21.0",
+    "fs-extra": "^5.0.0",
+    "lodash": "^4.17.5",
     "sinon": "^2.0.0-pre.5"
```
Comment (yjpa7145) [inline cumulus/tasks/discover-pdrs/package.json]: This can be removed.

File: docker-compose.yml [hunk 2]
```diff
@@ -21,13 +21,12 @@ services:
     ports:
       - "8080:80"
     volumes:
-      - "./packages/test-data:/var/www/html"
+      - "./tmp-test-data:/var/www/html"
```
Comment (yjpa7145) [inline docker-compose.yml]: Make this `:ro`

File: gibs-dashboard/app/scripts/config/production-generated.js [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (yjpa7145) [inline gibs-dashboard/app/scripts/config/production-generated.js]: What is this?

Comment (yjpa7145) [reply gibs-dashboard/app/scripts/config/production-generated.js]: This file can be deleted.

File: packages/common/test-utils.js [hunk 4]
```diff
@@ -107,3 +112,110 @@ async function createQueue() {
...
+ *
+ * Takes the same parameters as fs.readFile:
+ *
+ * https://nodejs.org/docs/v6.10.3/api/fs.html#fs_fs_readfile_file_options_callback
+ *
+ * @param {string|Buffer|integer} file - filename or file descriptor
+ * @param {any} options - encoding and flag options
+ * @returns {Promise} - the contents of the file
+ */
+function promisedReadFile(file, options) {
```
Comment (yjpa7145) [inline packages/common/test-utils.js]: Can this be removed?

File: packages/ingest/pdr.js [hunk 5]
```diff
@@ -395,6 +257,9 @@ class HttpParse extends httpMixin(baseProtocol(Parse)) {}
  *
  * @class
  */
+<<<<<<< HEAD
```
Comment (yjpa7145) [inline packages/ingest/pdr.js]: Fix this.

File: tmp-test-data/.gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+*
```
Comment (yjpa7145) [inline tmp-test-data/.gitignore]: Move this directory to `.tmp-test-data`

File: cumulus/tasks/discover-pdrs/package.json (hunk 1 above)
Comment (yjpa7145) [reply cumulus/tasks/discover-pdrs/package.json]: Fixed.

File: cumulus/tasks/discover-pdrs/index.js [hunk 7]
```diff
@@ -17,78 +16,60 @@ const local = require('@cumulus/common/local-helpers');
 function discoverPdrs(event) {
   try {
     const config = get(event, 'config', {});
-    const queue = get(config, 'useQueue', true);
```
Comment (kkelly51) [inline cumulus/tasks/discover-pdrs/index.js]: Why are you removing all of the Queue stuff?

File: packages/ingest/pdr.js (hunk 5 above)
Comment (kkelly51) [reply packages/ingest/pdr.js]: Please.

File: cumulus/tasks/discover-pdrs/index.js (hunk 7 above)
Comment (yjpa7145) [reply cumulus/tasks/discover-pdrs/index.js]: When I added output schema validation, I realized that we were returning completely different responses depending on whether or not queuing was used.  Not only was that causing the tests to now fail, it was also confusing for people trying to use that task.  I talked it over with @scisco and we decided that:

1) The discover tasks were doing to too much.  They were both discovering granules or PDRs, but they also contained the logic to then enqueue those items for ingest.
2) There are already `queue-granules` and `queue-pdrs` tasks which do a fine job of enqueuing items to be ingested.
3) We should remove the queuing functionality from the discovery tasks, simplifying those tasks.

So, that's why all of the queuing functionality has been removed from granule and PDR discovery.  And also a big part of reason that this PR is gigantic.

File: packages/ingest/pdr.js (hunk 5 above)
Comment (yjpa7145) [reply packages/ingest/pdr.js]: Done.

File: tmp-test-data/.gitignore (hunk 6 above)
Comment (yjpa7145) [reply tmp-test-data/.gitignore]: Done.

File: cumulus/tasks/discover-granules/tests/index.js [hunk 8]
```diff
@@ -78,6 +78,12 @@ test('discover granules using SFTP', async (t) => {
     t.is(output.granules.length, 3);
     t.is(output.granules[0].files.length, 2);
   }
+  catch (err) {
+    if (err.code === 'ECONNREFUSED') {
```
Comment (kkelly51) [inline cumulus/tasks/discover-granules/tests/index.js]: Yay! good.

**Pull request merged**